### ISSUE TITLE
Dragging windows is broken since 2.4.0

### ISSUE THREAD

**Issue by ElvenSpellmaker** [mintty/mintty#566]
Since 2.4.0 on Windows 8.1 dragging windows onto my second desktop from the third causes them to jolt around, enlarge the font and the size of the terminal and won't move more than halfway over my second desktop (so I cant drag them to the first desktop of three).

If I use the Windows key to move a window to the first desktop and then try dragging to the third, then the terminal gets larger (and zooms in) with each monitor I pass.

I'd like to say my middle (second) monitor is via USB3.0 --> HDMI if that makes a difference.

Note I am _not_ pressing any keys while moving the window.

**Comment by mintty** [comment]
Please fix the version you mentioned (2.2.4?).
What are the dimensions of your monitors? (Pixels, physical size, configured DPI values/"zooming factors"). Still running Windows 8.1?
Any change with option HandleDPI=false?

**Comment by ElvenSpellmaker** [comment]
17.3" Laptop 1920x1080 (1)
2x 23" HP Monitors 1920x1080 (2 and 3)

Therefore monitor 1 is denser (and everything appears smaller because Windows sucks at DPI Scaling).
But this problem happens when dragging from any monitor to any other monitor, even when the DPI is the same.

Yes I'm still running 8.1.

![image](https://cloud.githubusercontent.com/assets/2286713/16759576/9499c612-480f-11e6-90d0-f085a5e90da2.png)

Putting the option `HandleDPI=false` into my config solves the problem, and so it seems that the option as true (the default?) is broken?

**Comment by mintty** [comment]
So do I understand right? If you drag mintty from one external monitor to the other, both window and font size get larger? And if you drag it back to the previous external monitor, it gets larger again?
What are the individual DPI values/zoom factors of both (or all three) monitors?
Which mintty version has the effect?

**Comment by ElvenSpellmaker** [comment]
Yes that's right, plus it won't let me move it from 3 to anywhere over the second half of screen 2, it just keeps jumping back. I could perhaps try to get a video?

All monitors are on 100%:
![image](https://cloud.githubusercontent.com/assets/2286713/16762064/d5673854-481a-11e6-9db4-d01a8129decc.png)

The latest mintty started this (2.4.0).

**Comment by mintty** [comment]
Your initial report starts "Since 1.2.4..."; what's 1.2.4?
A video might help debugging with a chance of < 10% :(
No idea how to reproduce this.

**Comment by ElvenSpellmaker** [comment]
Sorry it's a mistype, I forgot the real version number! I'll try to get a video soon.

**Comment by ElvenSpellmaker** [comment]
I tried to use a screen recorder but it just goes wild on this laptop, and so I resorted to using my tablet's camera (which is pretty bad) as my phone camera is ruined.

https://www.youtube.com/watch?v=JcLElyf6SWA

Hopefully this shows what I mean. =)

**Comment by mintty** [comment]
The snapping effects are discomforting; I have not seen anything like it in my own multi-monitor experiments. Some comments:

* This does not seem to be a scaling problem as in some other issues. Font size is only slightly affected.
* Quite at the beginning, the window snaps up considerably; is it resized on that occasion?

**Comment by FredDeschenes** [comment]
I'm seeing the same issue, also running Windows 8.1 and Mintty 2.4.0.

If it helps both my monitors run the same resolution (1680X1050) with the same scaling.

Also the video posted by @ElvenSpellmaker doesn't really show it, but going from monitor 2 (on the right) to monitor 1 (left), I can get the "resize" thing to happen more than once if I keep dragging the window towards the left.

**Comment by ElvenSpellmaker** [comment]
@FredDeschenes: You're right, I was moving it slowly for the demonstration, but if you move it quickly the font enlarges once for every jerk back to the previous screen.

And yes, I can move with the Win+Arrows too. =)

**Comment by ElvenSpellmaker** [comment]
If it helps, the terminal resizes so that it still has the same number of lines and columns as before the resize of the text.

**Comment by mintty** [comment]
> If it helps, the terminal resizes so that it still has the same number of lines and columns as before the resize of the text.

which is a tweak that I introduced on purpose in 2.4.0 to stabilize the terminal size after DPI adaptation.
There are actually 3 related modifications; if you like you could try to disable them to find the culprit:
heuristic attempt to stabilize font size roundtrips, esp. after fulls… · mintty/mintty@236ea3a
tweak DPI changes to stabilize roundtrips of font and window size (#470) · mintty/mintty@2c3c081

**Comment by FredDeschenes** [comment]
@mintty : Reverting 2c3c0814143db085daed5ce188bb7b4798d8c56d in the 2.4.0 tag seems to have fixed the issue for me, thanks!

**Comment by mintty** [comment]
There are actually two changes in this commit:
```
-        long width = (r->right - r->left);
-        long height = (r->bottom - r->top);
+        long width = (r->right - r->left) * 20 / 19;
+        long height = (r->bottom - r->top) * 20 / 19;
```
and
```
+        int y = term.rows, x = term.cols;
         win_adapt_term_size(false, true);
+        win_set_chars(y, x);  // also clips to desktop size (win_fix_position)
```
Can you test them separately?

**Comment by FredDeschenes** [comment]
By itself, removing both " * 20 / 19"s fixes the issue, but the "mouse weirdness" (for the lack of a better word) is still there (cursor doesn't stay in the same place on the title bar dragging across screens). This is fixed by commenting the call to "win_set_chars".

**Comment by FredDeschenes** [comment]
Yeah that's really weird, even if they weren't the same DPI (which they might be now that I think of it), I really don't see how this could stack up for each monitor switch.

**Comment by mintty** [comment]
Please check repository version.

**Comment by FredDeschenes** [comment]
Looks good, thanks for the fix!

**Comment by ElvenSpellmaker** [comment]
Seems good to me!

**Comment by cdjc** [comment]
This issue has been bothering me for a couple of weeks, but finally looked in to it today, and good to see it already fixed now. Thanks. I've reverted back to 2.3.7 (via Cygwin) for the time being.

**Comment by ElvenSpellmaker** [comment]
@cdjc: Did you try the suggestion by @mintty? 

`HandleDPI=false` on 2.4.0?

**Comment by cdjc** [comment]
@ElvenSpellmaker: Just re-upgraded to 2.4.0 and tried it. Yes, it does solve the problem.

**Comment by mintty** [comment]
Fixed in 2.4.1.

**Comment by c-sonntag** [comment]
This option `HandleDPI=false` it's ok for me !
Thank you @ElvenSpellmaker !

**Comment by mintty** [comment]
Actually, with 2.4.1, this workaround should not be needed anymore.

**Comment by mintty** [comment]
With support for Windows DPI handling V2 (#774), this should now work without DPI disabling.

**Comment by mintty** [comment]
Released 2.9.0.
